src/o365/pa/commands/solution/solution-reference-add.ts

```diff
@@ -0,0 +1,148 @@
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  path: string;
+}
+
+/*
+ * Logic extracted from bolt.module.solution.dll
+ * Version: 1.0.6
+ * Class: bolt.module.solution.verbs.SolutionAddReferenceVerb
+ */
+class PaSolutionReferenceAddCommand extends Command {
+  public get name(): string {
+    return commands.SOLUTION_REFERENCE_ADD;
+  }
+
+  public get description(): string {
+    return 'Adds a reference inside the project in the current directory to the project at \'path\'';
+  }
+
+  public action(): CommandAction {
+    const cmd: Command = this;
+    return function (this: CommandInstance, args: CommandArgs, cb: (err?: any) => void) {
+      args = (cmd as any).processArgs(args);
+      (cmd as any).initAction(args, this);
+      cmd.commandAction(this, args, cb);
+    }
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: (err?: any) => void): void {
+    try {
+      const referencedProjectFilePath = this.getSupportedProjectFiles(args.options.path)[0];
+      const relativeReferencedProjectFilePath = path.relative(process.cwd(), referencedProjectFilePath);
+      const cdsProjectFilePath = this.getCdsProjectFile(process.cwd())[0];
+      var cdsProjectFileContent = fs.readFileSync(cdsProjectFilePath, 'utf8');
+
+      var cdsProjectMutator = new CdsProjectMutator(cdsProjectFileContent);
+      cdsProjectMutator.addProjectReference(relativeReferencedProjectFilePath);
+
+      fs.writeFileSync(cdsProjectFilePath, cdsProjectMutator.cdsProjectDocument);
+
+      cb();
+    }
+    catch (err) {
+      cb(new CommandError(err));
+    }
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-p, --path <path>',
+        description: 'The path to the referenced project.'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(): CommandValidate {
+    return (args: CommandArgs): boolean | string => {
+
+      var existingCdsProjects = this.getCdsProjectFile(process.cwd());
+      if (existingCdsProjects.length === 0) {
+        return 'CDS solution project file with extension cdsproj was not found in the current directory.';
+      }
+      if (existingCdsProjects.length > 1) {
+        return 'Multiple CDS solution project files with extension cdsproj were found in the current directory.';
+      }
+
+      if (!args.options.path) {
+        return 'Missing required option path.';
+      }
+
+      if (!fs.existsSync(args.options.path)) {
+        return `Path ${args.options.path} is not a valid path.`;
+      }
+
+      var existingSupportedProjects = this.getSupportedProjectFiles(args.options.path);
+      if (existingSupportedProjects.length !== 1) {
+        if (existingSupportedProjects.length === 0) {
```

Comment:
We can simplify the code by taking the `length === 0` comparison out and check for it first.